src/utils.jl

```diff
@@ -38,23 +38,85 @@ function _get_label_plot(gp::GraphPlot, labels)
     end
 end
 
+"""
+    getedgeinds(gr::G, edgedat::D) where {G<:AbstractGraph, K<:AbstractEdge, D<:AbstractDict{K}, IsDirected{G}}
+
+Return enumeration of edges for directed graph
+"""
+@traitfn function getedgeinds(gr::G, edgedat::D) where {G<:AbstractGraph, K<:AbstractEdge, D<:AbstractDict{K}; IsDirected{G}}
+    return edges(gr)
+end
+
+"""
+    getedgeinds(gr::G, edgedat::D) where {G<:AbstractGraph, K<:AbstractEdge, D<:AbstractDict{K}, IsDirected{G}}
+
+Return enumeration of edges for undirected graph such that the user's keys are used
+"""
+@traitfn function getedgeinds(gr::G, edgedat::D) where {G<:AbstractGraph, K<:AbstractEdge, D<:AbstractDict{K}; !IsDirected{G}}
+    Iterators.map(e -> reverse(e) ∈ keys(edgedat) ? reverse(e) : e , edges(gr))
```

Comment:
I don't get that part. Wouldn't that mean that a graph
```
g = SimpleDiGraph(2)
add_edge!(g, 2, 1)
```
could be colored using
```
ecolor = Dict(SimpleEdge(1,2)=>:red)
```
just because `1->2` happens to be present in the provided dict? Is that desired? I mean in directed graphs `1->2` and `2->1` are just different objects and i would expect to hit the default path then.

Reply:
Currently your example will produce a black edge as expected
```julia
graphplot(g, edge_color=Dict(Edge(2,1)=>:red))
```
![image](https://github.com/MakieOrg/GraphMakie.jl/assets/27723218/3386240d-e223-4df0-937a-f9dec1819d5e)

You can think about it like that: In case of undirected graphs we modify the `edges()` function such that it uses the same naming as the user used in the provided `Dict`. So for every element in `edges()` it will use either the one as the user identified the specific edge or the default from the `edges()`

For undirected graphs always the [`IsDirected`](https://github.com/MakieOrg/GraphMakie.jl/pull/89/files#diff-47c27891e951c8cd946b850dc2df31082624afdf57446c21cb6992f5f4b74aa2R46) function is called which always returns exactly `edges()`

I still think it works as intended. 
Would you agree now ? :)

I can add an extended help doc for the developers, as for the user it should be intuitive.

Reply:
Hah, sorry for that. I didn't pay enought attantion. That codepath is hit in case of **undirected** graphs while I was thinking it is hit in case of **directed** graphs where it doesn't make sense. The differens a `!` can make...